Makefile

```diff
@@ -451,6 +453,47 @@ st-checks:
 	# running on the host.
 	iptables-save | grep -q 'calico-st-allow-etcd' || iptables $(IPT_ALLOW_ETCD)
 
+## Get the kubeadm-dind-cluster script
+tests/k8st/dind-cluster.sh:
+	wget -O tests/k8st/dind-cluster.sh https://raw.githubusercontent.com/neiljerram/kubeadm-dind-cluster/master/dind-cluster.sh
+	chmod +x tests/k8st/dind-cluster.sh
+
+.PHONY: k8s-test
+## Run the k8s tests
+k8s-test: $(NODE_CONTAINER_CREATED) calico_test.created tests/k8st/dind-cluster.sh k8s-stop k8s-start
+	docker run \
+	    -v $(CURDIR):/code \
+	    -v /var/run/docker.sock:/var/run/docker.sock \
+	    -v /home/$(USER)/.kube/config:/root/.kube/config \
+	    -v /home/$(USER)/.kubeadm-dind-cluster:/root/.kubeadm-dind-cluster \
+	    --privileged \
+	    --net host \
+        $(TEST_CONTAINER_NAME) \
+	    sh -c 'cp /root/.kubeadm-dind-cluster/kubectl /bin/kubectl && ls -ltr /bin/kubectl && which kubectl && cd /code/tests/k8st && nosetests -v --with-xunit --xunit-file="/code/report/k8s-tests.xml" --with-timer'
+	$(MAKE) k8s-stop
```

Comment:
Do we want to do this as part of the test target?

It would be really useful if a test fails to be able to poke around the cluster afterwards.

Reply:
I think mainline is for the test to run and clean up after itself.  It's easy enough for a developer to comment that line when they want to poke around.  WDYT?

Reply:
I think we have a DEBUG_FAILURES flag defined for our other e2es - can we base it off that?

I'm not a huge fan of workflows that rely on people knowing which bits of code to comment out.